lib/pbench/test/unit/server/test_user_management_cli.py

```diff
@@ -0,0 +1,257 @@
+
+            result = runner.invoke(user_delete, args=[TestUserManagement.USER_TEXT])
+            assert result.exit_code == 0
+            assert result.stdout == "User test_user deleted\n"
+            assert not result.stderr_bytes
+
+    @staticmethod
+    @responses.activate
+    def test_invalid_user_delete(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            # Give username that doesn't exists to delete
+            result = runner.invoke(user_delete, args=["wrong_username"])
+            assert result.exit_code == 1
+            assert result.stdout == "User wrong_username does not exists\n"
+            assert not result.stderr_bytes
+
+    @staticmethod
+    @responses.activate
+    def test_valid_user_list(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            TestUserManagement.test_args(server_config, pytestconfig)
+
+            result = runner.invoke(user_list,)
+            row_format = "{0:15}\t{1:15}\t{2:15}\t{3:15}\t{4:20}"
+            assert result.exit_code == 0
+            assert (
+                result.stdout
+                == row_format.format(
+                    "Username", "First Name", "Last Name", "Registered On", "Email"
+                )
+                + "\n"
+                + row_format.format(
+                    TestUserManagement.USER_TEXT,
+                    TestUserManagement.FIRST_NAME_TEXT,
+                    TestUserManagement.LAST_NAME_TEXT,
+                    datetime.datetime.now().strftime("%Y-%m-%d"),
+                    TestUserManagement.EMAIL_TEXT,
+                )
+                + "\n"
+            )
+            assert not result.stderr_bytes
+
+    @staticmethod
+    @responses.activate
+    def test_valid_user_update(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            TestUserManagement.test_args(server_config, pytestconfig)
+
+            # Update email field
+            result = runner.invoke(
+                user_update,
+                args=[
+                    TestUserManagement.USER_TEXT,
+                    TestUserManagement.EMAIL_SWITCH,
+                    "new_test@domain.com",
+                ],
+            )
+            assert result.exit_code == 0
+            assert result.stdout == "User test_user updated\n"
+            assert not result.stderr_bytes
+
+            # Update valid role of the user
+            result = runner.invoke(
+                user_update,
+                args=[
+                    TestUserManagement.USER_TEXT,
+                    TestUserManagement.ROLE_SWITCH,
+                    "ADMIN",
+                ],
+            )
+            assert result.exit_code == 0
+            assert result.stdout == "User test_user updated\n"
+            assert not result.stderr_bytes
```

Comment:
There are some ways to parameterize a single test so that it gets run repeatedly with different values.  It seems like that would be worth considering here, as it would allow you to easily and compactly test all the options (and, if you include the exit code, et al. in the parameters, you could include the invalid-role scenario, as well).

Reply:
Yes, e.g., `@pytest.mark.parametrize("keys", ({value1}, ...))` which I've used in several of my tests; it's very handy, and quite flexible.